reproducibility/codalab/reproduce.py

```diff
@@ -0,0 +1,605 @@
+                        f"wilds:{wilds_src_uuid}",
+                        f"{dataset_fullname}:{dataset_uuid}",
+                        self._construct_command(
+                            dataset,
+                            algorithm="ERM",
+                            seed=0,
+                            hyperparameters=hyperparameter_config,
+                        ),
+                    ],
+                )
+
+    def output_hyperparameters_tuning_results(
+        self, worksheet_uuid, datasets, split, experiment_name=None
+    ):
+        self._set_worksheet(worksheet_uuid)
+
+        # For each dataset, output the run bundle info with the best metric value for a given split
+        for dataset in datasets:
+            metrics = get_metrics(dataset)
+            metric = metrics[0]
+            print(f"Using {metric} for {split} to find the best hyperparameters...\n")
+
+            # If experiment_name is not specified just default to ID naming scheme
+            if not experiment_name:
+                experiment_name = f"hyperparameters_id_erm_{dataset}"
+
+            grid_uuids = self._run(
+                [
+                    "cl",
+                    "search",
+                    experiment_name,
+                    "state=ready",
+                    f"host_worksheet={worksheet_uuid}",
+                    ".limit=50",
+                    "--uuid-only",
+                ]
+            ).split("\n")
+            best_uuid = None
+            best_metric_value = 0
+            for uuid in grid_uuids:
+                # CodaLab returns an empty string if there are no results
+                if not uuid:
+                    continue
+
+                results_dfs = load_results(
+                    f"https://worksheets.codalab.org/rest/bundles/{uuid}/contents/blob",
+                    splits=["val"],
+                    include_in_distribution=True,
+                )
+                val_result_df = get_early_stopped_row(
+                    results_dfs[split], results_dfs[split], metrics
+                )
+                if val_result_df[metric] > best_metric_value:
+                    print(
+                        f"Found a new best {val_result_df[metric]} > {best_metric_value}. UUID: {uuid}"
+                    )
+                    best_uuid = uuid
+                    best_metric_value = val_result_df[metric]
+
+            if best_uuid:
+                bundle_name = self._get_field_value(best_uuid, "name")
+                bundle_description = self._get_field_value(best_uuid, "description")
+
+                print("\n")
+                print("-" * 100)
+                print(
+                    f"Best for {dataset} \nuuid: {best_uuid} \nName: {bundle_name} \nDescription: {bundle_description}"
+                )
+                print("-" * 100)
+                print("\n")
+            else:
+                print(f"\nDid not find ready bundles for {dataset}.")
+
```

Comment:
I used this to output the full results, which I then used to update the tables in the paper.

Reply:
Ok thanks! Hm, so this gets printed out, and then you have to manually copy them into `output_full_results`?

Reply:
After I see which bundle has the best values, I rerun it with different seed values. After those runs finish, I then call output_full_results and pass in the worksheet uuid and experiment name to output full results. I can add logic behind a flag that auto reruns replicates.